src/router/Router.jsx

```diff
@@ -10,15 +10,44 @@ import {
   SignUp,
   WatchLater,
 } from "../pages/pages";
+import RequiresAuth from "./utils/RequiresAuth";
 
 export default function Router() {
   return (
     <Routes>
       <Route path="/" element={<Explore />} />
-      <Route path="/history" element={<History />} />
-      <Route path="/liked" element={<Liked />} />
-      <Route path="/playlist" element={<Playlist />} />
-      <Route path="/watchlater" element={<WatchLater />} />
+      <Route
+        path="/history"
+        element={
+          <RequiresAuth>
+            <History />
+          </RequiresAuth>
+        }
+      />
```

Comment:
you can use outlet in your requireAuth provider it will reduce the line of code,

Reply:
I just checked it out, this is great! Thank you!